### PR TITLE
Typo fixes; numbering fixes.

### PR DESCRIPTION
**Pull request by kitography**
Set of typos I noticed when looking at : https://github.com/hashicorp/vault/pull/14996

### CODE REVIEW

File: website/content/docs/secrets/kv/kv-v2.mdx [hunk 1]
```diff
@@ -185,7 +185,7 @@ real path).
    version          1
    ```
 
-1. Read arbitrary data:
+2. Read arbitrary data:
```
Comment (stevendpclark) [inline website/content/docs/secrets/kv/kv-v2.mdx:188]: I believe we don't need to update the numbers here, as the markdown generation will take care of incrementing the numbers properly.

Comment (kitography) [reply website/content/docs/secrets/kv/kv-v2.mdx]: ^ You are correct, is updated to remove that.

**Review by stevendpclark — APPROVED** [review]
👍 LGTM!

**Review by raskchanky — APPROVED** [review]

**Comment by kitography** [comment]
fmt failure is addressed here: https://github.com/hashicorp/vault/pull/15011
test-go-race failure is a flaky test (in list for the collab session to look at).

**Pull request merged**